classes/rest/class-rest-api.php

```diff
@@ -170,17 +173,26 @@ public static function endpoints(): void {
 				[
 					'methods'             => WP_REST_Server::CREATABLE,
 					'callback'            => static function ( WP_REST_Request $request ) {
-						$blocks      = [
+						$blocks     = [
 							SplitTwoColumns::BLOCK_NAME => SplitTwoColumns::class,
 						];
-						$block_name  = $request->get_param( 'blockname' );
+						$block_name = $request->get_param( 'blockname' );
+						/**
+						 * @var Base_Block $block_class
+						 */
 						$block_class = $blocks[ $block_name ] ?? null;
 
-						return rest_ensure_response(
-							$block_class && method_exists( $block_class, 'update_data' )
-								? $block_class::update_data( $request->get_params() )
-								: new WP_Error( 'error', 'Unknown block ' . ( $block_name ?? 'unspecified' ) )
-						);
+						if ( null === $block_class ) {
+							return new WP_Error( 'error', 'Unknown block ' . ( $block_name ?? 'unspecified' ) );
```

Comment:
We don't want  `rest_ensure_response()` on this one ? Just asking because we use it on the other WP_Error via `$response`; `rest_ensure_response` triggers one more action through `is_wp_error()`, but in the end returns the error as well :shrug:

Reply:
It's not needed, as the first thing line of `rest_ensure_response()` checks if the argument is a `WP_Error` and returns the argument itself if so https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api.php#L563-L565

But it's probably more straightforward to have the catch clause return the error as well. :+1: